File: app/src/main/java/com/example/mvvm_artichecture_sample/data/MainRepositoryHandler.java
package com.example.mvvm_artichecture_sample.data;

import com.example.mvvm_artichecture_sample.data.remote.apimodel.ReposnseModel;

import io.reactivex.Observable;
import retrofit2.Response;

public interface MainRepositoryHandler {
    Observable<Response<ReposnseModel>> getList();
}

File: app/src/main/java/com/example/mvvm_artichecture_sample/data/MainRepositoryHandler.kt
package com.example.mvvm_artichecture_sample.data

import com.example.mvvm_artichecture_sample.base.network.Output
import com.example.mvvm_artichecture_sample.data.remote.apimodel.ResponseModel


interface MainRepositoryHandler {
    suspend fun countries(requestType:String): Output<ResponseModel>

}
